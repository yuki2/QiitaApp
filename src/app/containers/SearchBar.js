import React, { PureComponent } from 'react';
import { StyleSheet, View, TextInput } from 'react-native';

import { PRIMARY_COLOR } from '../design';

const styles = StyleSheet.create({
  container: {
    backgroundColor: PRIMARY_COLOR,
    height: 40,
    padding: 5,
  },
  textInput: {
    height: 30,
    paddingTop: 5,
    paddingBottom: 5,
    paddingLeft: 20,
    textAlign: 'left',
    borderColor: '#444',
    backgroundColor: '#f7f7f7',
    borderRadius: 5,
    fontSize: 13,
  },
});

type Props = {
  onChangeText: (text: string) => void,
};
export default class SearchContainer extends PureComponent<Props> {
  render() {
    // inside your render function
    return (
      <View style={styles.container}>
        <TextInput
          style={styles.textInput}
          onChangeText={this.props.onChangeText}
          placeholder="Search"
          underlineColorAndroid="transparent"
        />
      </View>
    );
  }
}
