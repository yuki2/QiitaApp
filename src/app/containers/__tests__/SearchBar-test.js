import React from 'react';
import renderer from 'react-test-renderer';
import SearchBar from '../SearchBar';

it('renders correctly', () => {
  const onChangeText = () => {};
  const tree = renderer.create(<SearchBar onChangeText={onChangeText} />).toJSON();
  expect(tree).toMatchSnapshot();
});
