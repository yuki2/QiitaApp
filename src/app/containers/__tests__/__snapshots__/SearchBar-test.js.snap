// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#59BB0C",
      "height": 40,
      "padding": 5,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Search"
    style={
      Object {
        "backgroundColor": "#f7f7f7",
        "borderColor": "#444",
        "borderRadius": 5,
        "fontSize": 13,
        "height": 30,
        "paddingBottom": 5,
        "paddingLeft": 20,
        "paddingTop": 5,
        "textAlign": "left",
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
